### PR TITLE
[DEMO] upload, get and delete files manually

### LINKED ISSUE #1645: [DEMO] entity fields refactoring
As I see, for now we have at least **2** different names for single entity, it's a `toolName` and `projectName`. While in terms of Github it's called in third way - `repository`

May be we can open an issue and unify this terms one time?

That's why I asked in other conversation about relation to the `github`, because `tool` associated to me with some executable and not `repository`

_Originally posted by @kgevorkyan in https://github.com/saveourtool/save-cloud/pull/1644#discussion_r1053449820_

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
* Fixed `/upload-file` endpoint
* Added `/delete-file` endpoint both on backend (for permission check) and on demo
* Added `/list-file` endpoint both on backend (for permission check) and on demo
* Renamed fields of demo entities to match common practice (`organizationName` and `projectName`) (which closes #1645)

### CODE REVIEW

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt [hunk 1]
```diff
@@ -87,6 +94,64 @@ class ManagementController(
         .flatMap {
             demoService.loadFileToStorage(organizationName, projectName, version, file)
         }
+        .map {
+            FileDto(
```
Comment (nulls) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt:98]: `FileDto` supposes to store this key in database and use it as key of storage.
why do you need `FileDto` here? you have `ToolKey`

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/storage/ToolStorage.kt [hunk 2]
```diff
@@ -22,5 +22,5 @@ class ToolStorage(
         return ToolKey(ownerName, toolName, vcsTagName, executableName)
     }
 
-    override fun buildPathToContent(rootDir: Path, key: ToolKey): Path = rootDir / key.ownerName / key.toolName / key.vcsTagName / key.executableName
+    override fun buildPathToContent(rootDir: Path, key: ToolKey): Path = rootDir / key.organizationName / key.projectName / key.version / key.fileName
```
Comment (nulls) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/storage/ToolStorage.kt:25]: suggestion to store keys in database and use primary ID as path. It avoids problem with renaming and escaping a spec symbols in file name

Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/storage/ToolStorage.kt:25]: Will be done when the frontend part of file uploading is done

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt:98]: For now with use `ToolKey`, will be changed in PR with db-storage integration

Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt:98]: Oh will leave as is as far as `ToolKey` is `save-demo` specific

**Pull request merged**